Server/src/main/java/org/openas2/cmd/processor/restapi/AuthenticationFilter.java

```diff
@@ -0,0 +1,120 @@
+import java.util.Set;
+import java.util.StringTokenizer;
+import javax.annotation.security.DenyAll;
+import javax.annotation.security.PermitAll;
+import javax.annotation.security.RolesAllowed;
+import javax.ws.rs.container.ContainerRequestContext;
+import javax.ws.rs.container.ResourceInfo;
+import javax.ws.rs.core.Context;
+import javax.ws.rs.core.MultivaluedMap;
+import javax.ws.rs.core.Response;
+import javax.ws.rs.ext.Provider;
+import org.bouncycastle.util.encoders.Base64;
+
+/**
+ *
+ * @author javier
+ */
+@Provider
+public class AuthenticationFilter implements javax.ws.rs.container.ContainerRequestFilter {
+
+    @Context
+    private ResourceInfo resourceInfo;
+
+    private static final String AUTHORIZATION_PROPERTY = "Authorization";
+    private static final String AUTHENTICATION_SCHEME = "Basic";
+    private static final Response ACCESS_DENIED = Response.status(Response.Status.UNAUTHORIZED)
+            .entity("You cannot access this resource").build();
+    private static final Response ACCESS_FORBIDDEN = Response.status(Response.Status.FORBIDDEN)
+            .entity("Access blocked for all users !!").build();
+    private String adminUsername;
+    private String adminPassword;
+
+    public AuthenticationFilter(String adminUsername, String adminPassword) {
+        this.adminUsername = adminUsername;
+        this.adminPassword = adminPassword;
+    }
+
+    @Override
+    public void filter(ContainerRequestContext requestContext) {
+        Method method = resourceInfo.getResourceMethod();
+        //Access allowed for all
+        if (!method.isAnnotationPresent(PermitAll.class)) {
+            //Access denied for all
+            if (method.isAnnotationPresent(DenyAll.class)) {
+                requestContext.abortWith(ACCESS_FORBIDDEN);
+                return;
+            }
+
+            //Get request headers
+            final MultivaluedMap<String, String> headers = requestContext.getHeaders();
+
+            //Fetch authorization header
+            final List<String> authorization = headers.get(AUTHORIZATION_PROPERTY);
+
+            //If no authorization information present; block access
+            if (authorization == null || authorization.isEmpty()) {
+                requestContext.abortWith(ACCESS_DENIED);
+                return;
+            }
+
+            //Get encoded username and password
+            final String encodedUserPassword = authorization.get(0).replaceFirst(AUTHENTICATION_SCHEME + " ", "");
+
+            //Decode username and password
+            String usernameAndPassword = new String(Base64.decode(encodedUserPassword.getBytes()));
+
+            //Split username and password tokens
+            final StringTokenizer tokenizer = new StringTokenizer(usernameAndPassword, ":");
+            final String username = tokenizer.nextToken();
+            final String password = tokenizer.nextToken();
+
+            //Verifying Username and password
+            System.out.println(username);
```

Comment:
Please use Logger for logging

Reply:
Agreed - I only raised the PR to see the state of the branch nd perhaps pull some of the owork in but will not be merging this brnach.